src/main.ts

```diff
@@ -52,6 +52,9 @@ export async function main() {
         inlineScript = ` set -e >&2; echo '${START_SCRIPT_EXECUTION_MARKER}' >&2; ${inlineScript}`;
         scriptFileName = await createScriptFile(inlineScript);
 
+        const azureConfigDir = process.env.AZURE_CONFIG_DIR || path.join(process.env.HOME, '.azure'); 
+        process.env.AZURE_CONFIG_DIR = '/root/.azure';

```

Comment:
This changes the `AZURE_CONFIG_DIR` env var of the host.

If any Azure CLI command is run after `docker run` is executed, it will see `AZURE_CONFIG_DIR` as `/root/.azure` which doesn't exist on the host.

I would recommend exactly setting `AZURE_CONFIG_DIR` env var of the docker process via [`-e`, `--env`](https://docs.docker.com/reference/cli/docker/container/run/#env), instead of inheriting from `process.env`.

Reply:
No Azure CLI command will run after executing `docker run`. According to the [Node.js doc](https://nodejs.org/api/process.html#processenv), `process.env` changes do not affect the environment outside the Node.js process or other worker threads. Setting `process.env.AZURE_CONFIG_DIR = '/root/.azure'`; only impacts the current process and is not passed to subsequent processes.

Using the `-e` flag to set `AZURE_CONFIG_DIR` would complicate the code, as it requires omitting `AZURE_CONFIG_DIR` on the host and then explicitly mounting it. I don't think this approach is more efficient than resetting `AZURE_CONFIG_DIR` within the current process.

Reply:
The most efficient way is to not passing `AZURE_CONFIG_DIR` to the container at all.